### ISSUE TITLE
Const definitions with arithmetic operations fail in `verus!` macro

### ISSUE THREAD

**Issue by matthias-brun** [secure-foundations/verus#217]
Only happens in the new `verus!` macro. Arithmetic operations in const definitions promote the number to nat or int.

```rust
#[allow(unused_imports)]
use builtin::*;
use builtin_macros::*;

verus! {

const X: usize = 10 * 10;

fn main() {
}
}
```

```
fhs m: source ν ./tools/rust-verify.sh /st/verus/test.rs
error[E0308]: mismatched types
 --> /st/verus/test.rs:7:18
  |
7 | const X: usize = 10 * 10;
  |                  ^^^^^^^ expected `usize`, found struct `builtin::nat`

error: aborting due to previous error

For more information about this error, try `rustc --explain E0308`.
verification results:: verified: 0 errors: 0
```
cc @Chris-Hawblitzel

**Comment by Chris-Hawblitzel** [comment]
Should be fixed by 114a4b4cee7337bcaa67b6923b75ae0083674503

**Comment by utaal** [comment]
I'm going to close this, @matthias-brun feel free to reopen if you need to
